source/Calamari.Common/Features/Scripting/Bash/BashScriptBootstrapper.cs

```diff
@@ -44,6 +45,9 @@ public static string PrepareConfigurationFile(string workingDirectory, IVariable
             var encryptedVariables = EncryptVariables(variables);
             builder.Replace("#### VariableDeclarations ####", string.Join(LinuxNewLine, GetVariableSwitchConditions(encryptedVariables)));
 
+            builder.Replace("#### BashParametersArrayFeatureToggle ####", FeatureToggle.BashParametersArrayFeatureToggle.IsEnabled(variables) ? "true" : "false");
+            builder.Replace("#### VariableNamesArrayDeclarations ####", FeatureToggle.BashParametersArrayFeatureToggle.IsEnabled(variables) ? string.Join(" ", GetVariableNameAndValueDeclaration(encryptedVariables)) : string.Empty);
```

Comment:
We only replace the variableNamesArrayDecleration if the feature toggle is enabled. This prevents any malformed variables from breaking the bash script if the toggle is disabled.